src/api/recordlinker/main.py

```diff
@@ -83,13 +76,20 @@ async def not_found_handler(request, exc):
             return FileResponse(os.path.join(settings.ui_static_dir, "404.html"), status_code=404)
         raise exc
 
-    @app.get("/")
+    # page routes
     @app.get("/wizard")
-    @app.get("/favicon.ico")
     async def page(request: fastapi.Request):
         """
         Route to handle custom HTML pages for the UI
         """
         path: str = request.url.path.strip("/") or "index"
         view: str = path if path.endswith(".ico") else f"{path}.html"
         return FileResponse(os.path.join(settings.ui_static_dir, view))
+
+    # Other static files of NextJS app (includes index.html)
+    # [page-rehydration].txt files, images and js/css bundles in _next
+    app.mount(
+        "/",
+        StaticFiles(directory=os.path.join(settings.ui_static_dir), html = True),
+        name="SpaStatic",
+    )
```

Comment:
@ericbuckley I noticed that the path to the configuration files that instruct how to do page hydration (This is what allows the multiple pages to behave as SPA after first load) was broken.

I have mounted the root of the nextJS output to `/` but this time I have configured the index.html to be served as part of the same mount configuration, I hope this alleviates the concern of the same path being setup in multiple ways.

Reply:
@johanna-skylight are those the index.txt and wizard.txt files?

My biggest concern was that we have multiple routes that will match the same URL pattern (eg `/wizard` matches both the routes on line 80 and 91) and is there any chance that fastapi will randomly select 1 over the other when that happens.  From what I can tell it's not random, and fastapi is using the route that was registered first, so I don't want to hold this up.

I'm going to add an issue for me to investigate this a little more in the future, but I believe this is something we can come back to.

Reply:
Yes, I am referring to the index.txt and wizard.txt. Regarding `multiple routes that will match the same URL pattern (eg /wizard)`, the app.mount in line 91 does not resolve html pages by default. The only page that it resolves when setting `http = True` is `/` to `index.html`. It does not resolved any other paths to their html equivalents. 

You can test this in local. If you comment out the wizard route you will see the page will no longer load. This is why I started defining the page routes independently. I hope this clears the last concern.